lib/kafka/producer_interceptors.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of Kafka::ProducerInterceptor
+  # and wraps calls to a chain of custom producer interceptors.
+  class ProducerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This method is called when the client produces a message, before it gets queued to be sent to the Kafka brokers.
+    # The message returned from the first ProducerInterceptor.on_send is passed to the second interceptor on_send, and so on in an
+    # interceptor chain. This method does not throw exceptions.
+    #
+    # @param value [String] the message data.
+    # @param key [String] the message key.
+    # @param headers [Hash<String, String>] the headers for the message.
+    # @param topic [String] the topic that the message should be written to.
+    # @param partition [Integer] the partition that the message should be written to.
+    # @param partition_key [String] the key that should be used to assign a partition.
+    # @param create_time [Time] the timestamp that should be set on the message.
+    #
+    # @return [PendingMessage] the intercepted message returned by the last interceptor.
+    def on_send(value:, key:, headers:, topic:, partition:, partition_key:, create_time:)
+      intercepted_value = value
+      intercepted_key = key
+      intercepted_headers = headers
+      intercepted_topic = topic
+      intercepted_partition = partition
+      intercepted_partition_key = partition_key
+      intercepted_create_time = create_time
+      @interceptors.each do |interceptor|
+        begin
+          intercepted_value, intercepted_key, intercepted_headers, intercepted_topic, intercepted_partition, partition_key, create_time = interceptor.on_send(
```

Comment:
I wonder if ruby with his dark magic allow to pass a variable as a extracted list of fields of the variable,
something like:
```ruby
msg = = interceptor.on_send(msg.asFieldsorAsHashYouGetTheIdea)
```

also, this line is too long and hard to read